### PR TITLE
feat: page listing Webhooks [WEB-211]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

At `/det/webhooks`, have a table listing all Webhooks. I modeled the page on ModelRegistry. There isn't pagination, but we could have client-side sort of columns.

We've decided it's "Webhooks" in code and UI text.

For our limited options (Default / Slack) (OnError / OnComplete) I am offering only one selection of these.

## Test Plan

In the DB: 

```sql
insert into webhooks (url, webhook_type) VALUES ('http://example.org', 'DEFAULT');
insert into webhooks (url, webhook_type) VALUES ('http://determined.ai', 'SLACK');

;;  get the id of the webhook created

insert into webhook_triggers (trigger_type, condition, webhook_id) VALUES ('EXPERIMENT_STATE_CHANGE', '{"state":"ERROR"}'::jsonb, 3);
```

- Log in as non-admin user. `/det/webhooks` should not show any webhooks.
- Log in as admin. `/det/webhooks` should display webhooks listed
- Test a webhook (I believe that backend throws an error on this)
- Delete a webhook
- Add a webhook using the create webhook button and modal menu

## Checklist
- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.
- [x] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

### CODE REVIEW

**Comment by mapmeld** [comment]
Current status of table and create modal:

Screen Shot 2022-10-10 at 5 14 55 PM

Screen Shot 2022-10-10 at 5 15 04 PM

File: webui/react/src/hooks/useModal/Webhook/useModalWebhookCreate.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+const useModalWebhookCreate = ({ onClose }: Props): ModalHooks => {
+  const [form] = Form.useForm<FormInputs>();
+  const { modalOpen: openOrUpdate, modalRef, ...modalHook } = useModal({ onClose });
+
+  const modalContent = useMemo(() => {
+    return (
+      <Form autoComplete="off" form={form} layout="vertical">
+        <Form.Item label="URL" name="url" rules={[{ message: 'URL is required ', required: true }]}>
```
Comment (julian-determined-ai) [inline webui/react/src/hooks/useModal/Webhook/useModalWebhookCreate.tsx:31]: We should add some validation for the `URL`.

**Review by julian-determined-ai — APPROVED** [review]
**non-blocking:** This PR does not add a way to navigate to the `webhooks` page. I am fine with either: 

1. Adding a navigation menu item to get to the webhooks page that is hidden behind a feature flag for now. 
2. Leaving the navigation for a follow up PR after the backend work has been completed and merged in.

**Comment by mapmeld** [comment]
Update: when `?f_webhooks=on` is in the URL, a link to Webhooks will appear in the navigation

Screen Shot 2022-10-13 at 11 45 03 AM

**Comment by mapmeld** [comment]
Closed because the PR needs to be from this repo and not from a fork

**Pull request closed without merging**